apps/aestratum/src/aestratum_config.erl

```diff
@@ -0,0 +1,195 @@
+     {max_workers, ?DEFAULT_MAX_WORKERS}];  %% Per connection.
+defaults_schema(reward) ->
+    [beneficiaries,
+     keys,
+     {reward_last_rounds, ?DEFAULT_REWARD_LAST_ROUNDS}].
+
+
+section_map(Section, Cfg) ->
+    config_map(defaults_schema(Section), maps:from_list(val(Section, Cfg, []))).
+
+
+configure(RawConfigMap) ->
+    try lists:foldl(fun configure/2, RawConfigMap, section_keys()) of
+        #{} = Result ->
+            {ok, Result}
+    catch
+        _:{error, _} = ErrReason -> ErrReason
+    end.
+
+configure(_Section, {error, Reason}) ->
+    {error, Reason};
+configure(connection, #{connection := #{max_connections := MaxConnections,
+                                        num_acceptors := NumAcceptors,
+                                        transport := Transport} = ConnCfg,
+                        session := #{extra_nonce_bytes := ExtraNonceBytes}} = Result) ->
+    check_extra_nonce_bytes(MaxConnections, NumAcceptors, ExtraNonceBytes)
+        orelse error(insufficent_extra_nonce_bytes),
+    maps:merge(maps:without([connection], Result),
+               maps:put(transport, binary_to_atom(Transport, utf8), ConnCfg));
+configure(session, #{session := #{desired_solve_time := DesiredSolveTime,
+                                  max_solve_time := MaxSolveTime} = SessionCfg} = Result) ->
+    SessionCfg1 = SessionCfg#{desired_solve_time => DesiredSolveTime * 1000,
+                              max_solve_time => MaxSolveTime * 1000},
+    maps:merge(maps:without([session], Result), SessionCfg1);
+configure(reward, #{reward := #{beneficiaries := PoolShareBins,
+                                reward_last_rounds := LastN,
+                                keys := [{<<"dir">>, KeysDir}]}} = Result) ->
+    (is_integer(LastN) andalso LastN > 0 andalso LastN < 10)
+        orelse error({invalid, reward_last_rounds}),
+
+    ContractAddr = case aec_governance:get_network_id() of
+                       <<"ae_mainnet">> -> ?PAYMENT_CONTRACT_MAINNET_ADDRESS;
+                       _Other -> ?PAYMENT_CONTRACT_TESTNET_ADDRESS
+                   end,
+    ContractPK   = aestratum_conv:contract_address_to_pubkey(ContractAddr),
+
+    {CallerPK, CallerSK} = CallerKeyPair = read_keys(KeysDir, await),
+    CallerAddr   = aestratum_conv:account_pubkey_to_address(CallerPK),
+    check_keypair_roundtrips(CallerKeyPair) orelse error(invalid_keypair),
+
+    PoolPercentShares =
+        lists:foldl(fun (Bnf, Acc) ->
+                            [<<"ak_", _/binary>> = AccountAddr, PctShareBin] =
+                                binary:split(Bnf, <<":">>),
+                            PctShare = aestratum_conv:binary_to_number(PctShareBin),
+                            Acc#{AccountAddr => maps:get(AccountAddr, Acc, 0) + PctShare}
+                    end, #{}, PoolShareBins),
+    PoolPercentSum = lists:sum(maps:values(PoolPercentShares)),
+    PoolPercentSum < 100.0 orelse error(beneficiaries_reward_too_high),
```

Comment:
I would expect `=<`?

Reply:
Fixed. Although it would be a very nasty pool which would take away absolutely ALL mining rewards.